lib/components/Masthead/Masthead.tsx

```diff
@@ -72,150 +76,123 @@ export interface MastheadProperties {
     };
     toolbarItems?: Array<MastheadToolbarItem>;
     user?: React.ReactNode;
+    attr?: MastheadAttributes;
 }
 
-// Root container
-const MastheadContainer = styled(Attr.div)`
-    &&&& {
-        color: ${(props: ThemeProps<ShellTheme>) => props.theme.colorTextMastheadRest};
-        background-color: ${(props: ThemeProps<ShellTheme>) => props.theme.colorBgMasthead };
-    }
-`;
-
-// Translates Shell theme to action button theme under current context.
-function toolbarButtonTheme(theme: ShellTheme): ShellTheme {
-    return theme ? {
-        base: theme.base,
-        colorBgBtnStandardRest: theme.colorBgMasthead,
-        colorBgBtnStandardHover: theme.colorBgMastheadHover,
-        colorBgBtnStandardDisabled: theme.colorBgMastheadDisabled,
-        colorTextBtnStandardRest: theme.colorTextMastheadRest,
-        colorTextBtnStandardDisabled: theme.colorTextMastheadDisabled
-    } : { base: 'light' }; // Theme must be an object.
-}
-
-const StyledButton = styled(Attr.button)`
-    &&&&& {
-        color: ${(props: ThemeProps<ShellTheme>) => props.theme.colorTextBtnStandardRest};
-        &:hover { 
-            background-color: ${(props: ThemeProps<ShellTheme>) => props.theme.colorBgBtnStandardHover};
-        }
-    }
-`;
-
-export class Masthead extends React.PureComponent<MastheadProperties> {
-
-    getToolbarItems = () => {
-        if (!this.props.toolbarItems) {
-            return null;
-        }
-        return this.props.toolbarItems.map((item, idx) => {
-            const { label, icon, onClick, selected, attr } = item;
-            return (
-                <li key={idx} className={cx('masthead-toolbar-btn-container', { 'selected-more': this.props.more.selected })}>
-                    <StyledButton
-                        title={label}
-                        attr={attr}
-                        onClick={onClick}
-                        className={cx('masthead-btn', 'masthead-toolbar-btn', { selected })}>
-                            <Icon icon={icon}>{label}</Icon>
-                    </StyledButton>
-                </li>
-            );
-        });
-    }
-
-    render() {
-        const {
-            navigation,
-            user,
-            search,
-            more,
-            logo,
-            branding
-        } = this.props;
-
-        const items = this.getToolbarItems();
-        const expanded = search && search.expanded;
-
+export const Masthead = React.memo((props: MastheadProperties) => {

```

Comment:
This change is not required for the colors migration, let's do it separately

Reply:
there's another PR for the whole masthead change